src/Pos/Update/MemState/Functions.hs

```diff
@@ -27,65 +25,19 @@ withUSLock action = do
     lock <- mvLock <$> askUSMemVar
     bracket_ (liftIO $ Lock.acquire lock) (liftIO $ Lock.release lock) action
 
--- | Modify MemPool using UpdatePayload and PollModifier.
---
--- UpdatePayload is used to add new data to MemPool. Data must be
--- verified by caller. It's added directly to MemPool.
---
--- PollModifier is used to remove or modify some data in MemPool.
--- All deleted proposals and votes for them are removed.
--- TODO [CSL-625] Deleted and modified votes from non-deleted proposals
--- are not handled properly.
-modifyMemPool :: UpdatePayload -> PollModifier -> MemPool -> MemPool
-modifyMemPool UpdatePayload {..} PollModifier{..} =
-     addModifiers . delModifiers . addProposal upProposal
+-- | Add given payload to MemPool.
+addToMemPool :: UpdatePayload -> MemPool -> MemPool
+addToMemPool UpdatePayload {..} = addProposal . addVotes
   where
-    delModifiers :: MemPool -> MemPool
-    delModifiers MemPool{..} = MemPool
-        (foldr' HM.delete mpProposals pmDelActiveProps)
-        (foldr' HM.delete mpLocalVotes pmDelActiveProps)
-
-    addModifiers :: MemPool -> MemPool
-    addModifiers MemPool{..} =
-        let removeNotPresentedInActiveProps :: UpId -> UpdateVotes -> UpdateVotes
-            removeNotPresentedInActiveProps id stVotes
-                | Just activeProposal <- HM.lookup id pmNewActiveProps =
-                    stVotes `HM.intersection` (psVotes activeProposal)
-                | otherwise = stVotes
-
-            filteredLocalVotes :: LocalVotes
-            filteredLocalVotes = HM.mapWithKey removeNotPresentedInActiveProps mpLocalVotes
-        in MemPool
-              (foldr' (uncurry HM.insert) mpProposals
-                  (HM.toList $ HM.map psProposal pmNewActiveProps))
-        (foldr' forceInsertVote filteredLocalVotes .
-             filter (memberVS pmNewActiveProps) $ upVotes)
-
-    addProposal :: Maybe UpdateProposal -> MemPool -> MemPool
-    addProposal Nothing  mp = mp
-    addProposal (Just p) MemPool {..} = MemPool
-        (HM.insert (hash p) p mpProposals)
-        mpLocalVotes
+    addProposal mp =
+        case upProposal of
+            Nothing -> mp
+            Just up -> mp {mpProposals = HM.insert (hash up) up (mpProposals mp)}
+    addVotes mp = mp {mpLocalVotes = foldr' forceInsertVote (mpLocalVotes mp) upVotes}
```

Comment:
Any reason for `foldr'` instead of `foldl'`? If you are purposefully inserting votes in one order and changing to a different order would break it, it'd be nice to add a comment here